serve/destination.go

```diff
@@ -0,0 +1,146 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			zerologLevel, err := zerolog.ParseLevel(logLevel.String())
+			if err != nil {
+				return err
+			}
+			var logger zerolog.Logger
+			if logFormat.String() == "json" {
+				logger = zerolog.New(os.Stdout).Level(zerologLevel)
+			} else {
+				logger = log.Output(zerolog.ConsoleWriter{Out: os.Stdout}).Level(zerologLevel)
+			}
+
+			// opts.Plugin.Logger = logger
+			var listener net.Listener
+			if network == "test" {
+				listener = bufconn.Listen(testBufSize)
+				testListener = listener.(*bufconn.Listener)
+			} else {
+				listener, err = net.Listen(network, address)
+				if err != nil {
+					return fmt.Errorf("failed to listen %s:%s: %w", network, address, err)
+				}
+			}
+			// See logging pattern https://github.com/grpc-ecosystem/go-grpc-middleware/blob/v2/providers/zerolog/examples_test.go
+			s := grpc.NewServer(
+				middleware.WithUnaryServerChain(
+					logging.UnaryServerInterceptor(grpczerolog.InterceptorLogger(logger)),
+				),
+				middleware.WithStreamServerChain(
+					logging.StreamServerInterceptor(grpczerolog.InterceptorLogger(logger)),
+				),
+			)
+
+			destination.plugin.SetLogger(logger)
+			pb.RegisterDestinationServer(s, &servers.DestinationServer{Plugin: destination.plugin})
+			version := destination.plugin.Version()
+
+			if destination.sentryDSN != "" && version != "development" {
+				err = sentry.Init(sentry.ClientOptions{
+					Dsn:              destination.sentryDSN,
+					Debug:            false,
+					AttachStacktrace: true,
+					Release:          destination.plugin.Version(),
+					// https://docs.sentry.io/platforms/go/configuration/options/#removing-default-integrations
+					Integrations: func(integrations []sentry.Integration) []sentry.Integration {
+						var filteredIntegrations []sentry.Integration
+						for _, integration := range integrations {
+							if integration.Name() == "Modules" {
+								continue
+							}
+							filteredIntegrations = append(filteredIntegrations, integration)
+						}
+						return filteredIntegrations
+					},
+				})
+				if err != nil {
+					log.Error().Err(err).Msg("Error initializing sentry")
+				}
+			}
+
+			logger.Info().Str("address", listener.Addr().String()).Msg("server listening")
+			if err := s.Serve(listener); err != nil {
+				return fmt.Errorf("failed to serve: %w", err)
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVar(&address, "address", "localhost:7777", "address to serve on. can be tcp: `localhost:7777` or unix socket: `/tmp/plugin.rpc.sock`")
+	cmd.Flags().StringVar(&network, "network", "tcp", `the network must be "tcp", "tcp4", "tcp6", "unix" or "unixpacket"`)
+	cmd.Flags().Var(logLevel, "log-level", fmt.Sprintf("log level. one of: %s", strings.Join(logLevel.Allowed, ",")))
+	cmd.Flags().Var(logFormat, "log-format", fmt.Sprintf("log format. one of: %s", strings.Join(logFormat.Allowed, ",")))
+	cmd.Flags().BoolVar(&noSentry, "no-sentry", false, "disable sentry")
```

Comment:
```suggestion
	cmd.Flags().BoolVar(&noSentry, "no-sentry", false, "disable Sentry telemetry")
```

Reply:
Should we maybe use `no-telemetry` rather than `no-sentry`? We may want to change the underlying provider at some point, and developers shouldn't need to care too much about what we are using

Reply:
👍 https://github.com/cloudquery/plugin-sdk/pull/54#discussion_r963330061

Reply:
I think it should be two different stuff as it's error-reporting and telemetry. another option `telemtry=error-reporting,all,none`

Reply:
I think `telemetry` usually means "anything that phones home", including metrics, logs and error reports. So `no-telemetry` would not send any data to us at all. That's probably granular enough for now, and we can introduce further options in the future if we feel it becomes necessary?

Reply:
Opened this instead https://github.com/cloudquery/plugin-sdk/issues/128